signer/local/local_test.go

```diff
@@ -107,6 +108,161 @@ func TestNewSignerFromFileNoUsageInPolicy(t *testing.T) {
+	if res != nil && err == nil {
+		t.Fatal("Incorrect inputs failed to produce correct results")
+	}
+
+	res, err = NewSignerFromFile("../../helpers/testdata/messedupcert.pem", "local.go", nil)
+	if res != nil && err == nil {
+		t.Fatal("Incorrect inputs failed to produce correct results")
+	}
+
+	res, err = NewSignerFromFile("../../helpers/testdata/cert.pem", "../../helpers/testdata/messed_up_priv_key.pem", nil)
+	if res != nil && err == nil {
+		t.Fatal("Incorrect inputs failed to produce correct results")
+	}
+}
+
+// test the private method
+func testSign(t *testing.T) {
+	signer, err := NewSignerFromFile("testdata/ca.pem", "testdata/ca_key.pem", nil)
+	if signer == nil || err != nil {
+		t.Fatal("Failed to produce signer")
+	}
+
+	pem, err := ioutil.ReadFile("../../helpers/testdata/cert.pem")
+	if err != nil {
+		t.Fatal("srsly?")
+	}
+	cert, err := helpers.ParseCertificatePEM(pem)
+	if err != nil {
+		t.Fatal("srsly?")
+	}
+
+	badcert := *cert
+	badcert.PublicKey = nil
+	profl := config.SigningProfile{Usage: []string{"FUCK", "YOU"}}
+	_, err = signer.sign(&badcert, &profl)
+
+	if err == nil {
+		t.Fatal("Improper input failed to raise an error")
+	}
+
+	// nil profile
+	_, err = signer.sign(cert, &profl)
+	if err == nil {
+		t.Fatal("Nil profile failed to raise an error")
+	}
+
+	// empty profile
+	_, err = signer.sign(cert, &config.SigningProfile{})
+	if err == nil {
+		t.Fatal("Empty profile failed to raise an error")
+	}
+
+	// empty expiry
+	prof := signer.policy.Default
+	prof.Expiry = 0
+	_, err = signer.sign(cert, prof)
+	if err != nil {
+		t.Fatal("nil expiry raised an error")
+	}
+
+	// non empty urls
+	prof = signer.policy.Default
+	prof.CRL = "poop"
+	prof.OCSP = "poop"
+	prof.IssuerURL = []string{"poop"}
+	_, err = signer.sign(cert, prof)
+	if err != nil {
+		t.Fatal("non nil urls raised an error")
+	}
+
+	// nil ca
+	nilca := *signer
+	prof = signer.policy.Default
+	prof.CA = false
+	nilca.ca = nil
+	_, err = nilca.sign(cert, prof)
+	if err == nil {
+		t.Fatal("nil ca with isca false raised an error")
+	}
+	prof.CA = true
+	_, err = nilca.sign(cert, prof)
+	if err != nil {
+		t.Fatal("nil ca with CA true raised an error")
+	}
+}
+
+func TestSign(t *testing.T) {
+	testSign(t)
+	s, err := NewSignerFromFile("testdata/ca.pem", "testdata/ca_key.pem", nil)
+	if err != nil {
+		t.Fatal("Failed to produce signer")
+	}
+
+	//pem, _ := ioutil.ReadFile("../helpers/testdata/cert.pem")
+	//cert, _ := helpers.ParseCertificatePEM(pem)
+
+	// test the empty request
+	_, err = s.Sign(signer.SignRequest{})
+	if err == nil {
+		t.Fatalf("Empty request failed to produce an error")
+	}
+
+	pem, err := ioutil.ReadFile("../../helpers/testdata/cert.pem")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	validReq := signer.SignRequest{Hostname: testHostName, Request: string(pem)}
+	_, err = s.Sign(validReq)
+	if err == nil {
+		t.Fatal("A bad case failed to raise an error")
+	}
+
+	validReq = signer.SignRequest{Hostname: "74.67.211.31", Request: string(pem)}
```

Comment:
I actually tried the hostname as IP thing here, but it still didn't touch that line.